repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -95,6 +100,14 @@ def get_events(pes_events_filepath):
         raise StopActorExecution()
 
 
+def filter_events_by_architecture(events, arch):
+    filtered_events = []
+    for event in events:
+        if not event.architectures or arch in event.architectures:
```

Comment:
I think you also need to add events with architecture '_noarch_'.

Reply:
There's no "noarch" architecture in the JSON. Events for all architectures have the field set to ["x86_64", "aarch64", "ppc64le", "s390x"].
(sorry for the delay)